src/import/pull-tar.c

```diff
@@ -315,11 +315,13 @@ static void tar_pull_job_on_finished(PullJob *j) {
         if (!tar_pull_is_done(i))
                 return;
 
-        if (i->checksum_job->style == VERIFICATION_PER_DIRECTORY && i->signature_job->error != 0) {
-                log_error_errno(j->error, "Failed to retrieve signature file, cannot verify. (Try --verify=no?)");
+        if (i->signature_job) {
+                if (i->checksum_job->style == VERIFICATION_PER_DIRECTORY && i->signature_job->error != 0) {
```

Comment:
According to pull-common.c a checksum job is always started unless verify is IMPORT_VERIFY_NO. A signature job is only started when verify is IMPORT_VERIFY_SIGNATURE in which case a checksum job is also started, because verify isn't IMPORT_VERIFY_NO. In other words, a signature job implies the presence of a checksum job, so I guess I only have to check for the signature job. 

Otherwise, I have updated my patch. Could you please review again?